docs/topics/Threads.md

```diff
@@ -0,0 +1,111 @@
+
+- `id`, `guild_id`, `type`, `name`, `last_message_id`, `last_pin_timestamp`, `rate_limit_per_user` are being re-used
+- `owner_id` has been repurposed to store the id of the user that started the thread
+- `parent_id` has been repurposed to store the id of the `GUILD_TEXT` or `GUILD_NEWS` channel the thread was created in
+
+Additionally, there are a few new fields that are only available on threads:
+
+- `message_count` and `member_count` store an approximate count, but they stop counting at 50 (these are only used in our UI, so likely are not valuable to bots)
+- `thread_metadata` contains a few thread specific fields, `archived`, `archive_timestamp`, `archiver_id`, `auto_archive_duration`, `locked`. Archive timestamp is changed when creating, archiving, or unarchiving a thread, and when changing the auto_archive_duration field.
+
+## Public & Private Threads
+
+Public threads are viewable by everyone who can view the parent channel of the thread. Public threads must be created from an existing message, but can be "orphaned" if that message is deleted. The [type](#DOCS_RESOURCES_CHANNEL/channel-object-channel-types) of thread created matches the [type](#DOCS_RESOURCES_CHANNEL/channel-object-channel-types) of the parent channel. `GUILD_TEXT` channels [create](#DOCS_RESOURCES_CHANNEL/start-a-public-thread) `PUBLIC_THREAD` and `GUILD_NEWS` channels [create](#DOCS_RESOURCES_CHANNEL/start-a-public-thread) `NEWS_THREAD`.
+
+Private threads behave similar to Group DMs, but in a Guild. Private threads are always [created](#DOCS_RESOURCES_CHANNEL/start-a-private-thread) with the `PRIVATE_THREAD` [type](#DOCS_RESOURCES_CHANNEL/channel-object-channel-types) and can only be created in `GUILD_TEXT` channels.
+
+## Active & Archived Threads
+
+Every thread can be either active or archived. Changing a thread from archived -> active is referred to as unarchiving the thread. Threads that have `locked` set to true can only be unarchived by a user with the `MANAGE_THREADS` permission.
+
+Besides helping to de-clutter the UI for users, archiving exists to limit the working set of threads that need to be kept around. Since the number of archived threads can be quite large, keeping all of them in memory may be quite prohibitive. Therefore guilds are capped at a certain number of active threads, and only active threads can be manipulated. Users cannot send messages, edit messages, add reactions, use slash commands, or join archived threads. The only operation that should happen within an archived thread is messages being deleted.
+
+Because of this constraint, the gateway protocol is designed to ensure that bots are able to have an accurate view of the full set of active threads, but archived threads are not synced up-front via the gateway.
+
+Threads do not count against the max-channels limit in a guild, but there will be a new limit on the maximum number of active threads in a guild.
```

Comment:
Don't think we need to, we don't expose other constants on the Guild object like the number of emojis a server can have?

Reply:
Ah, this is information we expose in JDA but I'm realizing it is because it is a hardcoded limited augmented by BoostTiers.
Part of this expectation was the thought that maximums might vary between Guilds which would be an important metric for Moderation bots to know in regards to handling auto-archives if needed or whatever else they might do to avoid hitting limits for various features.

Reply:
Gotcha.  If the limits become dynamic, we can make sure they're synced on the guild.  I suspect they will be fairly static though, and we'll just publish a few constants somewhere.